packages/cpsuLambda/tests/format_1/business_states.test.ts

```diff
@@ -1,26 +1,106 @@
-import {deliveryType, itemStatusType} from "../../src/schema/format_1"
-import {getBusinessStatus} from "../../src/schema/format_1/transformer"
-
-it("should convert a delivery type business status", () => {
-  const itemStatus: itemStatusType = "ReadyForCollection"
-  const deliveryType: deliveryType = "Robot Collection"
+import {BundleEntry, Task} from "fhir/r4"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {generateTemplate, getBusinessStatus, populateTemplate} from "../../src/schema/format_1/transformer"
+import {
+  itemType,
+  requestType,
+  deliveryType,
+  itemStatusType
+} from "../../src/schema/format_1"
 
+function testBusinessStatusConversion(itemStatus: itemStatusType, deliveryType: deliveryType, expectedStatus: string) {
   const businessStatus = getBusinessStatus(deliveryType, itemStatus)
-  expect(businessStatus.value()).toEqual("Ready to Collect")
-})
+  expect(businessStatus.value()).toEqual(expectedStatus)
+}
 
-it("should convert an item status business status", () => {
-  const itemStatus: itemStatusType = "NotDispensed"
-  const deliveryType: deliveryType = "Robot Collection"
+describe("getBusinessStatus function", () => {
+  it("should convert a delivery type business status", () => {
+    testBusinessStatusConversion("ReadyForCollection", "Robot Collection", "Ready to Dispatch")
+  })
 
-  const businessStatus = getBusinessStatus(deliveryType, itemStatus)
-  expect(businessStatus.value()).toEqual("Not Dispensed")
+  it("should convert an item status business status", () => {
+    testBusinessStatusConversion("NotDispensed", "Robot Collection", "Not Dispensed")
+  })
+
+  it("should return Nothing if the item status is not in the map", () => {
+    const itemStatus: itemStatusType = "Expired"
+    const deliveryType: deliveryType = "Robot Collection"
+
+    const businessStatus = getBusinessStatus(deliveryType, itemStatus)
+    expect(businessStatus.isNothing()).toBeTruthy()
+  })
+
+  it("should convert a DispensingComplete status with Robot Collection", () => {
+    testBusinessStatusConversion("DispensingComplete", "Robot Collection", "Dispatched")
+  })
+
+  it("should handle ReadyForCollection status with Robot Collection correctly", () => {
+    testBusinessStatusConversion("ReadyForCollection", "Robot Collection", "Ready to Dispatch")
+  })
+
+  it("should convert a delivery type business status for Delivery required", () => {
+    testBusinessStatusConversion("ReadyForCollection", "Delivery required", "Ready to Dispatch")
+  })
+
+  it("should convert an item status business status for Delivery required", () => {
+    testBusinessStatusConversion("DispensingComplete", "Delivery required", "Dispatched")
```

Comment:
Can we do these as one parameterised test instead of multiple tests that all just call the same function?

Reply:
The test has been parameterised.